### PR TITLE
doc: add proposal template.

### PR DESCRIPTION
**Pull request by SimonCqk**
### Ⅰ. Describe what this PR does

add proposal template doc.

### II. Does this pull request fix one issue?

NONE

### CODE REVIEW

File: .github/PROPOSAL_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+## Proposal
+
+## Use case
+
+## Implementation
+
+## Alternatives considered
+
+## Doc history
```
Comment (jian-he) [inline .github/PROPOSAL_TEMPLATE.md:15]: doc history can be removed ?

File: .github/PROPOSAL_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Proposal Template
+
+## Term
+
+## Motivations
+
+## Proposal
+
+## Use case
```
Comment (jian-he) [inline .github/PROPOSAL_TEMPLATE.md:9]: use case can be before proposal

File: .github/PROPOSAL_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Proposal Template
+
+## Term
```
Comment (jian-he) [inline .github/PROPOSAL_TEMPLATE.md:3]: Term can be removed for simplicity

File: .github/PROPOSAL_TEMPLATE.md (hunk 1 above)
Comment (SimonCqk) [reply .github/PROPOSAL_TEMPLATE.md]: done

**Pull request merged**